react-components/src/components/Reveal3DResources/Reveal3DResources.tsx

```diff
@@ -14,20 +25,109 @@ import {
   type TypedReveal3DModel,
   type AddResourceOptions
 } from './types';
+import { type CogniteExternalId } from '@cognite/sdk';
+import { type FdmAssetMappingsConfig, useFdmAssetMappings } from '../../hooks/useFdmAssetMappings';
+
+export type FdmAssetStylingGroup = {
+  fdmAssetExternalIds: CogniteExternalId[];
+  style: { cad?: NodeAppearance; pointcloud?: PointCloudAppearance };
+};
+
+export type Reveal3DResourcesStyling = {
+  defaultStyle?: { cad?: NodeAppearance; pointcloud?: PointCloudAppearance };
+  groups?: FdmAssetStylingGroup[];
+};
 
 export type Reveal3DResourcesProps = {
   resources: AddResourceOptions[];
+  fdmAssetMappingConfig?: FdmAssetMappingsConfig;
+  styling?: Reveal3DResourcesStyling;
 };
 
-export const Reveal3DResources = ({ resources }: Reveal3DResourcesProps): ReactElement => {
+export const Reveal3DResources = ({
+  resources,
+  styling,
+  fdmAssetMappingConfig
+}: Reveal3DResourcesProps): ReactElement => {
   const [reveal3DModels, setReveal3DModels] = useState<TypedReveal3DModel[]>([]);
+  const [reveal3DModelsStyling, setReveal3DModelsStyling] = useState<
+    Array<PointCloudModelStyling | CadModelStyling>
+  >([]);
+
   const { setModelsAdded } = useContext(ModelsLoadingStateContext);
   const viewer = useReveal();
   const numModelsLoaded = useRef(0);
 
+  const stylingExternalIds = useMemo(
+    () => styling?.groups?.flatMap((group) => group.fdmAssetExternalIds) ?? [],
+    [styling]
+  );
+
+  const { data: mappings } = useFdmAssetMappings(stylingExternalIds, fdmAssetMappingConfig);
+
   useEffect(() => {
     getTypedModels(resources, viewer).then(setReveal3DModels).catch(console.error);
-  }, [resources]);
+  }, [resources, viewer]);
```

Comment:
`viewer` in dependency?

Reply:
Yeah, I think following exhaustive dependencies rule when possible is a good thing. If, for some reason, new viewer is passed we need to update all models.